### PR TITLE
Q binning for non zero gamma angle

### LINKED ISSUE #28425: Change q binning for non zero gamma
When using ILLSANSAutoprocess, the q binnign can be determined automatically. But D16 gamma scans introduced data for which the detector is not in front of the beam. In these cases, for gamma above a certain value (~40°), the q binning calculation fails and enters an infinite loop, leading to Workbench freezing.

### PR DESCRIPTION
**Pull request by MathieuTillet**
**Description of work.**
This PR changes the behavior of the q binning algorithm in SANSILLIntegration when the detector is not in front of the beam (gamma angle different from 0).
After some tries, binning from q_min to q_max with step (q_max - q_min)/pixels_by_row was determined a reasonable calculation (the original way of computing the binning cannot be extended over the whole range of the instrument).

**To test:**
`SANSILLAutoProcessTest` has been updated to take this change into account. Alternatively, one can use the files used for testing D16 to check the results' coherence.

Fixes #28425 . 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by gemmaguest — CHANGES_REQUESTED** [review]
Looks good and the reduction still runs fine. Just one query.

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSILLIntegration.py [hunk 1]
```diff
@@ -217,7 +217,15 @@ def _get_iq_binning(self, q_min, q_max, pixel_size, wavelength, l2, binning_fact
                 q_binning = self._mildner_carpenter_q_binning(q_min, q_max, binning_factor)
             else:
                 if wavelength != 0:
-                    q_binning = self._pixel_q_binning(q_min, q_max, pixel_size * binning_factor, wavelength, l2, offset)
+                    run = mtd[self._input_ws].getRun()
+                    if run.hasProperty("Gamma.value") and run.getLogData("Gamma.value") != 0:
+                        if mtd[self._input_ws].getInstrument().getName() == "D16":
+                            pixel_nb = 320
+                        else:
+                            self.log().warning("Width of the instrument not found. Assuming 320 pixels.")
```
[the remark this answers is not in the document]
Comment (MathieuTillet) [reply Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSILLIntegration.py:225]: Fair point, I'll change that

**Comment by gemmaguest** [comment]
Does this require release notes?

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSILLIntegration.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (gvardany) [reply Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSILLIntegration.py:226]: Can you put the number of pixels in the instrument parameters file, and get from there?

[the remark this answers is not in the document]
Comment (gvardany) [reply Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSILLIntegration.py:226]: I would also swap the order of ifs, other instruments are not expected to have gamme defined, so this makes sense only for D16.

**Comment by gemmaguest** [comment]
> 
> 
> Does this require release notes?

@MathieuTillet @gvardany I'm happy to approve this, just wanted to check if it needs release notes?

**Comment by MathieuTillet** [comment]
Very sorry about this delay. Here are both release notes and the changes requested by Gagik.

**Review by gemmaguest — APPROVED** [review]
Thanks for the changes. Still runs fine and the value is picked up from the params file or logs a warning if it's not there :+1:

**Pull request merged**